### PR TITLE
Jrg4017/update pyton errors jg

### PR DESCRIPTION
**Pull request by jrgabler**
@Gladdstone resolving py errors so the frontend server can run successfully

### CODE REVIEW

File: src/Resources/MarketProvider.py [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
\ No newline at end of file
+    # def getStock(self, symbol: str) -> Stock:
+    #     response = requests.get(API_URL + 'markets/quotes',
+    #     params={'symbols': symbol, 'greeks': 'false'},
+    #     headers={'Authorization': API_TOKEN, 'Accept': 'application/json'}
+    #
+    #     # throws an error because you need to pass an asset
+    #     stock = Stock()
+    #     json.dumps(stock)
+    #     return stock
```
Comment (jrgabler) [inline src/Resources/MarketProvider.py:21]: dunno what need to happen here but commenting out

File: src/api/StockService.py [hunk 2]
```diff
@@ -1,8 +1,8 @@
 from flask_restful import Resource, reqparse
 from flask_jwt_extended import (create_access_token, create_refresh_token, jwt_required, jwt_refresh_token_required, get_jwt_identity, get_raw_jwt)
 
-from Controllers import StockController
-from Models.assets import Stock
+from controllers import StockController
+from models.assets import Stock
```
Comment (jrgabler) [inline src/api/StockService.py:5]: files seem to need to match capitalization of folder names.

File: src/api/UserService.py [hunk 3]
```diff
@@ -1,25 +1,25 @@
...
 
-from Controllers import UserController
-from Models import User
+from controllers import UserController
+from models import User
 
 parser = reqparse.RequestParser()
 parser.add_argument("username", help="This field cannot be blank", required=True)
-parser.add_argument("password", help="This field cannot be blank", require=True)
+parser.add_argument("password", help="This field cannot be blank", required=True)
```
Comment (jrgabler) [inline src/api/UserService.py:9]: `require` -> `required`

File: src/controllers/StockController.py [hunk 4]
```diff
@@ -86,7 +88,12 @@ def purchaseAsset(self, stock: Stock, user: User):
...
-            cursor.execute(f"INSERT INTO Stock VALUES({stock.open}, {stock.close}, {stock.high}, {stock.low}, {stock.average_volume}, {stock.dividend_yield}, {stock.type}, {stock.last}, {stock.symbol}, {stock.prevclose})")
             # there's a lot to do here
             # we need to get the purchase price
+            cursor.execute(f"INSERT INTO Stock VALUES({stock.open}, {stock.close}, {stock.high}, {stock.low}, {stock.average_volume}, {stock.dividend_yield}, {stock.type}, {stock.last}, {stock.symbol}, {stock.prevclose})")
+        except mysql.DatabaseError as error:
+            print(error) # TODO - log error
+        finally:
+            if(connection is not None):
+                connection.close()
+            return {"message": f"Purchased {stock.symbol}"}
```
Comment (jrgabler) [inline src/controllers/StockController.py:99]: added this as it was expecting both. Please update message @Gladdstone

File: src/main.py [hunk 5]
```diff
@@ -3,10 +3,10 @@
 from flask import Flask, render_template, redirect, url_for
 from flask import request
 from flask_restful import Api
-from flast_jwt_extended import JWTManager
+from flask_jwt_extended import JWTManager
```
Comment (jrgabler) [inline src/main.py:6]: `flast_...` -> `flask_...`

File: src/main.py [hunk 6]
```diff
@@ -21,16 +21,16 @@
 
 blacklist = set()
 
+jwt = JWTManager(app)
+
+api = Api(app)
+
 # called every time client tries to access a secure endpoint
 @jwt.token_in_blacklist_loader
```
Comment (jrgabler) [inline src/main.py:29]: need to declare jwt before using it in a func

File: src/models/assets/Stock.py [hunk 7]
```diff
@@ -1,9 +1,9 @@
 # import Asset
 
-# This object has to match the API being used in order to make json.dumps() 
+# This object has to match the API being used in order to make json.dumps()
 # work
-class Stock(Asset):
-    def __init__(self, open: float, close: float, high: float, low: float, average_volume: float, market_cap: float, peratio: float, dividend_yield: float, asset_type: string, last: float, symbol: string, prev_close: float):
+class Stock():
```
Comment (jrgabler) [inline src/models/assets/Stock.py:5]: removed `Asset` as it doesn't seem to be used for now. Please add when neccessary

`string` -> `str`

**Review by Gladdstone — APPROVED** [review]

**Pull request merged**